plugin/advisor/mysql/advisor_naming_foreign_key_convention.go

```diff
@@ -0,0 +1,161 @@
+package mysql
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/common"
+	"github.com/bytebase/bytebase/plugin/advisor"
+	"github.com/bytebase/bytebase/plugin/db"
+	"github.com/pingcap/tidb/parser/ast"
+)
+
+var (
+	_ advisor.Advisor = (*NamingFKConventionAdvisor)(nil)
+)
+
+func init() {
+	advisor.Register(db.MySQL, advisor.MySQLNamingFKConvention, &NamingFKConventionAdvisor{})
+	advisor.Register(db.TiDB, advisor.MySQLNamingFKConvention, &NamingFKConventionAdvisor{})
+}
+
+// NamingFKConventionAdvisor is the advisor checking for foreign key naming convention.
+type NamingFKConventionAdvisor struct {
+}
+
+// Check checks for foreign key naming convention.
+func (check *NamingFKConventionAdvisor) Check(ctx advisor.Context, statement string) ([]advisor.Advice, error) {
+	root, errAdvice := parseStatement(statement, ctx.Charset, ctx.Collation)
+	if errAdvice != nil {
+		return errAdvice, nil
+	}
+
+	level, err := advisor.NewStatusBySchemaReviewRuleLevel(ctx.Rule.Level)
+	if err != nil {
+		return nil, err
+	}
+
+	format, templateList, err := api.UnmarshalNamingRulePayloadAsTemplate(ctx.Rule.Type, ctx.Rule.Payload)
+	if err != nil {
+		return nil, err
+	}
+	checker := &namingFKConventionChecker{
+		level:        level,
+		format:       format,
+		templateList: templateList,
+	}
+	for _, stmtNode := range root {
+		(stmtNode).Accept(checker)
+	}
+
+	if len(checker.adviceList) == 0 {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    common.Ok,
+			Title:   "OK",
+			Content: "",
+		})
+	}
+
+	return checker.adviceList, nil
+}
+
+type namingFKConventionChecker struct {
+	adviceList   []advisor.Advice
+	level        advisor.Status
+	format       string
+	templateList []string
+}
+
+// Enter implements the ast.Visitor interface
+func (checker *namingFKConventionChecker) Enter(in ast.Node) (ast.Node, bool) {
+	indexDataList := checker.getMetaDataList(in)
+
+	for _, indexData := range indexDataList {
+		regex, err := getTemplateRegexp(checker.format, checker.templateList, indexData.metaData)
+		if err != nil {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    common.Internal,
+				Title:   "Internal error for foreign key naming convention rule",
+				Content: fmt.Sprintf("%q meet internal error %q", in.Text(), err.Error()),
+			})
+			continue
+		}
+		if !regex.MatchString(indexData.index) {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    common.NamingFKConventionMismatch,
+				Title:   "Mismatch foreign key naming convention",
+				Content: fmt.Sprintf("Foreign key mismatches the naming convention, expect %q but found `%s`", regex, indexData.index),
```

Comment:
Users can create a foreign key with no name and MySQL will automatically create a name for this situation. So that, we may get the error content like "Foreign key mismatches the naming convention, expect "fk_{table}_{column}" but found ``".
If there is more than one `CREATE TABLE` statement in a single task, users may get the multiple same error content as above. It confuses users and doesn't know which foreign key is referring to.

I suggest adding a specific table name here to distinguish, such as "Foreign key in table `xxx` mismatches ....."

Reply:
Make sense. I'll add the table name in the message